crates/sputnik/src/error.rs

```diff
@@ -0,0 +1,32 @@
+use thiserror::Error;
+
+use std::io;
+
+/// SputnikError is the type of Error that occured.
+#[derive(Error, Debug)]
+pub enum SputnikError {
+    /// IOError occurs when any given std::io::Error arises.
+    #[error(transparent)]
+    IOError(#[from] io::Error),
```

Comment:
I have the same scaling questions about the `from` implementations here. Is this better than asking the calling code to do the conversion? I'm really not sure!